new_dawn_server/users/api/resources.py

```diff
@@ -316,12 +316,32 @@ def _get_liker_info(self, bundle):
             
         if my_likes.count():
             bundle.data["liked_info_from_me"] = self._build_liker_dict(my_likes)
-            
-            
+
+
+    def _get_taken_info(self, bundle):
+        viewer_id = bundle.request.GET.get('viewer_id')
+        user_id = bundle.data["user"].data["id"]
+        your_taken = UserAction.objects.filter(
+            Q(user_from__id__exact=user_id)
+            & Q(user_to__id__exact=viewer_id)
+            & Q(action_type=ActionType.REQUEST_TAKEN.value)
+        )
+        my_taken = UserAction.objects.filter(
+            Q(user_from__id__exact=viewer_id)
+            & Q(user_to__id__exact=user_id)
+            & Q(action_type=ActionType.REQUEST_TAKEN.value)
+        )
+        if your_taken.count():
+            bundle.data["taken_requested"] = True
```

Comment:
Let's change this logic a bit and use "taken_requested_from_you" and "taken_requested_from_me" so the iOS can distinguish who is the request sender. The logic here can be: if you sent taken request to me, then "taken_requested_from_you" should be true. If I sent taken request to you, then "taken_requested_from_me" should be true. An example would be:
```
if your_taken.count():
            bundle.data["taken_requested_from_you"] = True
else:
      bundle.data["taken_requested_from_you"] = False
```
same for "taken_requested_from_me"